### PR TITLE
Improve pair programming guide

### PR DESCRIPTION
**Pull request by frahugo**

### CODE REVIEW

File: guides/practices/pair-programming.md [hunk 1]
```diff
@@ -73,3 +73,39 @@ It is the host's responsibility to ensure the following before the guest comes o
...
+The process is simple:
+
+1. A writes a new test and sees that it fails.
+2. B implements the code needed to pass the test.
+3. B writes the next test and sees that it fails.
+4. A implements the code needed to pass the test.
+
+And so on. **Refactoring is done whenever the need arises by whoever is driving**.
+
+- First, before either writing a test or implementing the code, a clear shared understanding has to be established.  Othewise, the watcher will have a hard time following the driver and disagreements may arise quickly
```
Comment (bokay) [inline guides/practices/pair-programming.md]: Small typo at "Otherwise"

File: guides/practices/pair-programming.md [hunk 2]
```diff
@@ -14,6 +22,7 @@ common set of tools with the same configurations:
 - [Atom](https://atom.io/); editor
 - [iTerm2](https://www.iterm2.com/); terminal
 - [SourceTree](https://www.sourcetreeapp.com/); Git GUI - optional
+- [Cloud9](https://aws.amazon.com/fr/cloud9/); pairing station in the cloud - remote only
 - [Zoom](https://zoom.us/); screen sharing - remote only
```
Comment (nicholasjhenry) [inline guides/practices/pair-programming.md]: Let's split the local and remote setups under the headers

File: guides/practices/pair-programming.md [hunk 3]
```diff
@@ -57,19 +66,73 @@ gracefully.
 
 ## Remote pairing
 
-Remote pairing requires a certain amount of preparation. Before starting the pairing session,
-determine who will be the "host". This is the developer who will be sharing their screen.
-It is the host's responsibility to ensure the following before the guest comes online:
+Rmeote pairing is done using a cloud pairing station. We use Cloud9. A pairing instance
```
Comment (nicholasjhenry) [inline guides/practices/pair-programming.md]: `Rmeote` -> `Remote`

File: guides/practices/pair-programming.md [hunk 4]
```diff
@@ -57,19 +66,73 @@ gracefully.
...
+must first be provisioned for the project. See [guide here](https://github.com/civilcode/cloud9-bootstrap).
+
+Both pair members access the Cloud9 instance using their Amazon IAM account. The pair use
+the tool they want for the audio (Zoom, slack, ...).
```
Comment (nicholasjhenry) [inline guides/practices/pair-programming.md]: Let's standardize on Zoom for audio as then it's easier to screenshare

File: guides/practices/pair-programming.md [hunk 5]
```diff
@@ -57,19 +66,73 @@ gracefully.
...
+
+## Git user
+
+For pulling and pushing to Github, we use our CivilCode pairing account (`civilcode-pairing`).
+The first time a repo is pulled through the `https` handle, git asks for the username and password.
+By default, the credentials are cached in OSX keychain.
+
+When committing code, the pair uses the [multiple co-authors](https://help.github.com/articles/creating-a-commit-with-multiple-authors/) feature of GitHub.
+The pairing station has a list of possible co-authors in the `.gitmessage` file.
```
Comment (nicholasjhenry) [inline guides/practices/pair-programming.md]: We should add them to the `dotfiles` project?

File: guides/practices/pair-programming.md [hunk 6]
```diff
@@ -57,19 +66,73 @@ gracefully.
...
+- The watcher should try to **limit interruptions** to prevent the driver loosing focus. The watcher should wait for a moment where an intervention is "a propos". The watcher is **not** a _navigator_ (concept from another pair programming process).
+- The watcher should not tell the driver what to do, unless the driver asks for it.
+- The driver regularly asks the watcher for feedback (errors? spelling mistakes? opinions?)
+- If the watcher wants to use the keyboard/mouse, it must first ask the driver.
+- The driver should not keep the keyboard for too long. Big refactoring can be split into smaller ones and each member in turn refactors portions of the code.
+- If the pair disagrees on design or implementation and cannot come up with an agreement, file an issue in the project and bring this up to the team at some point.
+- The code is shared by the development team, therefore use _we_ instead of _you_ or _I_ when referring to the code.
+- When pairing remotely, as a watcher, avoid distractions and stay in the same file as the driver, as if you were parigin in person.
```
Comment (nicholasjhenry) [inline guides/practices/pair-programming.md]: `parigin` -> `pairing`

File: guides/practices/pair-programming.md (hunk 5 above)
Comment (frahugo) [reply guides/practices/pair-programming.md]: Added issue to dotfiles project.

**Comment by nicholasjhenry** [comment]
@frahugo Looks like this as the paragraph that was deleted:

```
## Remote pairing

Remote pairing requires a certain amount of preparation. Before starting the pairing session,
determine who will be the "host". This is the developer who will be sharing their screen.
It is the host's responsibility to ensure the following before the guest comes online:
```

**Pull request merged**